apis/config/v1beta1/configuration_types.go

```diff
@@ -387,6 +387,25 @@ type Integrations struct {
 	LabelKeysToCopy []string `json:"labelKeysToCopy,omitempty"`
 }
 
+type IntegrationReference string
```

Comment:
Please extract this to a separate PR, as we agree on strong typing, but we are yet not sure about putting all references together.

Reply:
I can’t really decouple this. 

I can give a PR with just this type but it won’t build because we use the configiguretion list all over the repo.

Reply:
Can't we just use in these places the new type instead of string?

Reply:
to be more precise, just replace
```
defaultJobFrameworkName                             = "batch/job"
```
with 
```
defaultJobFrameworkName  kueue.IntegrationReference = "batch/job"
```
and so on in other places, not sure why this would not compile

Reply:
That is what this PR does.

Reply:
Here is the "minimal" change. https://github.com/kubernetes-sigs/kueue/pull/4697

In any place like jobframework that takes a string or is passed in our config list, I did a transformation to make it fit into a string array.

This allows me to avoid changing it all places but it does require me to convert it in a few places so we don't propogate this change everywhere.